pie/utils/objects.py

```diff
@@ -409,3 +409,127 @@ async def interaction_check(self, interaction: nextcord.Interaction) -> None:
         else:
             await super().interaction_check(interaction)
+
+
+class VoteEmbed(nextcord.ui.View):
+    """Class for making voting embeds easy.
+    The right way of getting response is first calling send() on instance,
+    then checking instance attribute `value`.
+
+    Attributes:
+        value: True if confirmed, False if declined, None if timed out
+        ctx: Context of command
+        message: Vote message
+
+    Args:
+        ctx: The context for translational and sending purposes.
+        embed: Embed to send.
+        limit: Minimal votes.
+        timeout: Number of seconds before timeout. `None` if no timeout
+        delete: Delete message after answering / timeout
+        vote_author: Auto vote for author
+
+
+    To use import this object and create new instance:
+    .. code-block:: python
+        :linenos:
+
+        from pie.utils.objects import VoteEmbed
+
+        ...
+
+        embed = utils.discord.create_embed(
+            author=reminder_user,
+            title=Vote for your action.",
+        )
+        view = VoteEmbed(ctx, embed)
+
+        value = await view.send()
+
+        if value is None:
+            await ctx.send(_(ctx, "Voted timed out."))
+        elif value:
+            await ctx.send(_(ctx, "Vote passed."))
+    """
+
+    def __init__(
+        self,
+        ctx: commands.Context,
+        embed: nextcord.Embed,
+        limit: int,
+        timeout: Union[int, float, None] = 300,
+        delete: bool = True,
+        vote_author: bool = False,
+    ):
+        super().__init__(timeout=timeout)
+        self.value: Optional[bool] = None
+        self.ctx = ctx
+        self.embed = embed
+        self.limit = limit
+        self.voted = []
+        self.delete = delete
+
+        if vote_author:
+            self.voted.append(ctx.author.id)
+
+    async def send(self):
+        """Sends message to channel defined by command context.
+        Returns:
+            True if confirmed in time, None if timed out
+        """
+
+        self.button = nextcord.ui.Button(
+            label=_(self.ctx, "Yes") + " ({})".format(len(self.voted)),
+            style=nextcord.ButtonStyle.green,
+            custom_id="yes-button",
+        )
+
+        self.add_item(self.button)
+
+        self.message = await self.ctx.send(embed=self.embed, view=self)
+        await self.wait()
+
+        if not self.delete:
+            self.clear_items()
+            await self.message.edit(embed=self.embed, view=self)
+        else:
+            try:
+                try:
+                    await self.message.delete()
+                except (
+                    nextcord.errors.HTTPException,
+                    nextcord.errors.Forbidden,
+                ):
+                    self.clear_items()
+                    await self.message.edit(embed=self.embed, view=self)
+            except nextcord.errors.NotFound:
+                pass
```

Comment:
Can't this be handled by one try-except block? Or, in this case, try-except-except?

Reply:
It's the same approach used in all Embed objects here. I think if we want to change this, we also have to change this behavior in all of them. And in that case it should be in it's own PR

Reply:
It could be just a separate commit. I'd like to have it the proper way from the start.

Reply:
I read the docs and handlig all Exceptions from delete simply by ignoring it is good enough, since if you get any error deleting messagen, you can't edit it either.